Mobile test

andresmadeddie opened the pull request:
Create SaucedemoMobilTest
create saucedemomobile package inside mobile directory this includes all common and android classes, so a util file
create two test classes, on to implement the util file
All method in the test were tested successfully

AndresSolvd committed:
change _config.properties for mobile test

AndresSolvd committed:
change _config.properties for mobile test

AndresSolvd committed:
change _config.properties for mobile test

andresmadeddie committed:
implement logIn test
pending xpath for productScreenTitle

andresmadeddie committed:
implement logIn test
pending xpath for productScreenTitle

andresmadeddie committed:
implement logIn test
pending xpath for productScreenTitle

AndresSolvd committed:
added login and addProduct tests

AndresSolvd committed:
added login and addProduct tests

AndresSolvd committed:
update

AndresSolvd committed:
update with tested productAddedToCartTest and longinTest

andresmadeddie committed:
Added two more test methods

andresmadeddie committed:
update

AndresSolvd committed:
update

AndresSolvd committed:
test all methods successfully

jinocencio-solvd commented on src/test/java/com/solvd/carina/demo/SauceDemoMobileTest.java:
```diff
@@ -0,0 +1,107 @@
+        ProductsScreenBase productsScreenPage = loginScreenBase.clickLoginButton();
+        Assert.assertTrue(productsScreenPage.isOpened(), "Products screen is not opened");
+    }
+    @Test
```
need empty line between test

jinocencio-solvd commented on src/test/java/com/solvd/carina/demo/SauceDemoMobileTest.java:
```diff
@@ -0,0 +1,107 @@
+    @Test
+    @MethodOwner(owner = "Andres")
+    public void productAddedToTheCartTest() {
+        String productTitle = R.TESTDATA.get("second_product_name");
```
Have you tried testing with your product_name = Sauce Labs Onesie? I think that product is further down the products screen

andresmadeddie replied:
Yes, To do it, it is necessary to swipe

jinocencio-solvd commented on src/test/java/com/solvd/carina/demo/SauceDemoMobileTest.java:
```diff
@@ -0,0 +1,107 @@
+        // Go to Cart screen and verify product has been added
+        CartScreenBase cartScreenBase = productDetailsScreenBase.clickCartButton();
+        Assert.assertTrue(cartScreenBase.isOpened(), "Cart screen is not opened");
+        Assert.assertTrue(cartScreenBase.isProductDisplayed(productTitle), "Product is not in the cart");
```
Have you tried more than 1 item added to cart?

asurelaba commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/AbstractSauceDemoScreen.java:
```diff
@@ -0,0 +1,14 @@
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = LoginScreenBase.class)
+public class AbstractSauceDemoScreen extends AbstractSauceDemoScreenBase {
+
```
This class is not used anywhere else. Am i missing something?

asurelaba commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/CartScreen.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Override
+    public CheckOutYourInformationScreenBase clickCheckOutButton() {
+        checkOutButton.click();
```
might have to swipe if there are many items in cart

asurelaba commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/LoginScreen.java:
```diff
@@ -0,0 +1,53 @@
+import java.lang.invoke.MethodHandles;
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = LoginScreenBase.class)
+public class LoginScreen extends LoginScreenBase implements IMobileUtils {
```
LoginScreenBase extends AbstractSauceDemoScreenBase which implements IMobileUtils. So, IMobileUtils can be removed here?

asurelaba commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/ProductDetailsScreen.java:
```diff
@@ -0,0 +1,52 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    //    @ExtendedFindBy(accessibilityId = "test-BACK TO PRODUCTS")
```
Commented code can be removed.

asurelaba commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/ProductsScreen.java:
```diff
@@ -0,0 +1,59 @@
+
+    @Override
+    public ProductDetailsScreenBase clickProductTitle(String productTitle) {
+        productTitleS.format(productTitle).click();
```
Would work only if product is at the top, for other products we would need to swipe to find them.

asurelaba reviewed (commented):
We would also need TestNG suite xml

ankochergin commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/CartScreen.java:
```diff
@@ -0,0 +1,52 @@
+            "android.view.ViewGroup/android.view.ViewGroup/" +
+            "android.view.ViewGroup/android.view.ViewGroup/android.view.ViewGroup/android.view.ViewGroup/" +
+            "android.view.ViewGroup/android.view.ViewGroup/android.view.ViewGroup[2]/android.widget.TextView")
+    private ExtendedWebElement screenTitle;
```
please avoid such long locators, this one looks like auto-generated. try to find some beautiful content-desc or find it by text

ankochergin commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/LeftSideMenu.java:
```diff
@@ -0,0 +1,39 @@
+    private ExtendedWebElement closeMenuButton;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-LOGOUT\"]/android.widget.TextView")
+    private ExtendedWebElement logOutButton;
```
same as in previous in comment

ankochergin commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/ProductsScreen.java:
```diff
@@ -0,0 +1,59 @@
+    @FindBy(xpath = "(//android.widget.TextView[@content-desc=\"test-Item title\"])[@text='%s']")
+    private ExtendedWebElement productTitleS;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-Menu\"]/android.view.ViewGroup/android.widget.ImageView")
```
```suggestion
    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-Menu\"]")
```

ankochergin commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/utils/AuthUtilsScreen.java:
```diff
@@ -0,0 +1,19 @@
+import com.zebrunner.carina.utils.factory.ICustomTypePageFactory;
+
+public class AuthUtilsScreen implements ICustomTypePageFactory{
+    public ProductsScreenBase login(String username, String password) {
```
```suggestion
public class AuthUtilsScreen implements ICustomTypePageFactory {

    public ProductsScreenBase login(String username, String password) {
```

ankochergin commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/utils/AuthUtilsScreen.java:
```diff
@@ -0,0 +1,19 @@
+import com.zebrunner.carina.utils.R;
+import com.zebrunner.carina.utils.factory.ICustomTypePageFactory;
+
+public class AuthUtilsScreen implements ICustomTypePageFactory{
```
I suppose it should not be screen, but just AuthUtil
```suggestion
public class AuthUtils implements ICustomTypePageFactory {
```

ankochergin replied:
If you have AuthUtils for web please consider renaming this class to MobileAuthUtils to avoid confusion

ankochergin commented on src/test/java/com/solvd/carina/demo/SauceDemoMobileTest.java:
```diff
@@ -0,0 +1,107 @@
+        ProductsScreenBase productsScreenBase = authUtilsScreen.loginStandardUser();
+        Assert.assertTrue(productsScreenBase.isOpened(), "Products screen is not opened");
+
+        // Go to Product screen
```
Usually we don't write comments in the tests, only if the logic is complicated. You have clear and understandable screens and methods name and we understand what your code is doing. I think comments can be removed

AndresSolvd committed:
Add a line between methods in SauceDemoMobileTest
Remove implementation on LoginScreen class as already existed in the current extended class
Added class in android.xml
Modified pointer hamburgerMenuButtonToGetLeftSideMenu
Added line in AuthUtilsScreen.java
Renamed to MobileAuthUtils
Remove comments in method checkOutProcessTest

AndresSolvd committed:
Add a line between methods in SauceDemoMobileTest
Remove implementation on LoginScreen class as already existed in the current extended class
Added class in android.xml
Modified pointer hamburgerMenuButtonToGetLeftSideMenu
Added line in AuthUtilsScreen.java
Renamed to MobileAuthUtils
Remove comments in method checkOutProcessTest
Modify xpath in LeftSideMenu
Modify cartScreen screenTitle xpath

AndresSolvd committed:
Remove all comments in SauceDemoMobileTest.java
change to @ExtendedFindBy in ProductDetailsScreen.java

andresmadeddie committed:
added swipe();

AndresSolvd committed:
Added swipe

AndresSolvd committed:
Merge remote-tracking branch 'origin/MobileTest' into MobileTest

AndresSolvd committed:
format change

AndresSolvd committed:
fix swipe()

andresmadeddie committed:
Merge branch 'master' into MobileTest

andresmadeddie committed:
delete unused class

ankochergin commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/CheckOutYourInformationScreen.java:
```diff
@@ -0,0 +1,57 @@
+    @FindBy(xpath = "//android.widget.EditText[@content-desc=\"test-Zip/Postal Code\"]")
+    private ExtendedWebElement zipCodeField;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-CONTINUE\"]/android.widget.TextView")
```
```suggestion
    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-CONTINUE\"]")
```
please check if this locator works

ankochergin commented on src/main/java/com/solvd/carina/demo/mobile/saucedemomobile/android/ProductsScreen.java:
```diff
@@ -0,0 +1,60 @@
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-Cart drop zone\"]//android.widget.TextView[@text='PRODUCTS']")
+    private ExtendedWebElement screenTitle;
+
+    @FindBy(css = "//android.view.ViewGroup[@content-desc='test-Cart']/android.view.ViewGroup/android.widget.ImageView")
```
```suggestion
    @FindBy(css = "//android.view.ViewGroup[@content-desc='test-Cart']")
```
please check if this locator works

AndresSolvd committed:
modify Xpath and tested successfully
